### PR TITLE
[OpenCL] Registers MaxPoolGrad and MaxPoolGradGrad

### PR DESCRIPTION
**Pull request by jeffmak**
@jwlawson I had to inline the atomic add functions because including sycl_util.h for more than one file gave me linking issues. Let me know if there are better ways to do this :)

### CODE REVIEW

File: tensorflow/core/kernels/maxpooling_op.cc [hunk 1]
```diff
@@ -966,6 +1110,255 @@ struct LaunchMaxPoolingGradGradWithArgmax<Eigen::GpuDevice, T> {
...
+        if (output_data[index] == input_data_n[idx]) {
+          maxidx = idx;
+          should_stop = true;
+        }
+      }
+    }
+    if (maxidx != -1) {
+      SyclAtomicAdd(
+          output_backprop +
+              n * p_.in_rows_ * p_.in_cols_ * p_.depth_ +
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/maxpooling_op.cc:1183]: can you run clang format on your code?

File: tensorflow/core/kernels/pooling_ops_common.h [hunk 2]
```diff
@@ -68,6 +68,69 @@ struct PoolParameters {
   TensorFormat data_format;
 };
 
+#ifdef TENSORFLOW_USE_SYCL
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/pooling_ops_common.h:71]: That will cause conflicts.. can you make PR for SYCL2DPoolParams on it's own? So we can merge that in ahead of the other changes?

File: tensorflow/core/util/sycl_util.h [hunk 3]
```diff
@@ -30,11 +30,11 @@ namespace tensorflow {
 // the header so that the host compiler can compile the functor.
 #ifdef __SYCL_DEVICE_ONLY__
 template <typename T>
-void SyclAtomicAdd(__attribute__((address_space(1))) T* address,
+inline void SyclAtomicAdd(__attribute__((address_space(1))) T* address,
```
Comment (lukeiwanski) [inline tensorflow/core/util/sycl_util.h:33]: That should be a PR on it's own. The smaller PR the better :)

**Review by lukeiwanski — CHANGES_REQUESTED** [review]
Overall looks good!

**Comment by jeffmak** [comment]
Closing PR, refer to #106.

**Pull request closed without merging**